app/lib/features/chat/widgets/chats_list.dart

```diff
@@ -185,17 +185,19 @@ class __AnimatedChatsListState extends State<_AnimatedChatsList> {
   void _insert(int pos, String data) {
     _log.fine('insert $pos: $data');
     _currentList.insert(pos, data);
-    if (_listKey.currentState != null) {
-      _listKey.currentState!.insertItem(pos);
+    final curState = _listKey.currentState;
+    if (curState != null) {
+      curState.insertItem(pos);
     } else {
       _log.fine('we are not');
     }
   }
 
   void _remove(int pos, String data) {
     _currentList.removeAt(pos);
-    if (_listKey.currentState != null) {
-      _listKey.currentState!.removeItem(
+    final curState = _listKey.currentState;
+    if (curState != null) {
+      curState.removeItem(
```

Comment:
The reason you are not using `.let` here is because there is some else? how about extending our `let` helper to take an optional `else:`-function that would be called?

(side note: why is it called `let` if it maps the value to the result of the function in the first place?)

Reply:
Yes, also I was hopping to define `else:` clause in `let` extension.
But I didn't know how to extend it.

Reply:
`Let` extension was designed by @gtalha07.
I thought he got inspired by `let` statement in rust language.
@gnunicorn If you have better idea in extension name, let's discuss together.

Reply:
nah, it is fine for this PR... I just noticed that it is a bit off ... as `map` is what it does ...

@gtalha07 what do you mean with it is reserved? that `.map` is used on lists/iterators? That's the same for rust... I don't see that as being a problem ... but our use of `.let` having side-effects feels quirky too me ... I might propose changing the idioms here a bit ...

Reply:
I could modify `Let` extension as following.
```dart
extension Let<T> on T? {
  R? let<R>(R? Function(T) op, {void Function()? orElse}) {
    final T? value = this;
    if (value != null) return op(value);
    if (orElse != null) orElse();
    return null;
  }

  // it supports async callback too unlike `extension_nullable`
  Future<R?> letAsync<R>(R? Function(T) op, {void Function()? orElse}) async {
    final T? value = this;
    if (value != null) return op(value);
    if (orElse != null) orElse();
    return null;
  }
}
```

Reply:
I suppose we could term it as ```option```? Since the type behaviour is derived from it ?? `map` isn't suitable as its dart reserved.

Reply:
I think there are many occurrences of `.let()` function that should be replaced with `orElse` clauser.
I am going to handle this issue in another PR.

Reply:
@gnunicorn I think @gtalha07 idea is correct.
Please think about case that we should wrap the original `.map()` by this extension.
If both are `.map()`, which one would be our extension?

Reply:
> `map` isn't suitable as its dart reserved.

no it isn't. Or I misunderstand what you mean by that. But if you checked https://github.com/acterglobal/a3/pull/2280 you'll see that using the name `map` isn't a problem at all..

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2274/commits/710566572d988d450628f56ef7342f6b622b819d

Reply:
@gnunicorn @gtalha07 Thank you for your earnest cooperation.